File: backend/src/main/java/me/performancereservation/global/security/KakaoOAuth2UserInfo.java
package me.performancereservation.global.security;

import java.util.Map;

public class KakaoOAuth2UserInfo extends OAuth2UserInfo {
    public KakaoOAuth2UserInfo(Map<String, Object> attributes) {
        super(attributes);
    }

    @Override
    public String getName() {
        Map<String, Object> account = (Map<String, Object>) attributes.get("kakao_account");
        Map<String, Object> profile = (Map<String, Object>) account.get("profile");

        return (String) profile.get("name");
    }

    @Override
    public String getEmail() {
        Map<String, Object> account = (Map<String, Object>) attributes.get("kakao_account");

        return (String) account.get("email");
    }

    @Override
    public String getOauthId() {
        return String.valueOf(attributes.get("oauthId"));
    }
}

File: backend/src/main/java/me/performancereservation/global/security/oauth/user/KakaoOAuth2UserInfo.kt
package me.performancereservation.global.security.oauth.user

class KakaoOAuth2UserInfo(attributes: Map<String, Any>) : OAuth2UserInfo(attributes) {
    override val oauthId: String
        get() = attributes["id"].toString()

    override val name: String
        get() =
        ((attributes["kakao_account"] as? Map<*, *>)?.get("profile") as? Map<*, *>)?.get("nickname") as? String ?: ""

    override val email: String get() =
        (attributes["kakao_account"] as? Map<*, *>)?.get("email") as? String ?: ""
}